File: backend/model/custom_training.go
package model

import "time"

type CustomTraining struct {
	ID               uint      `json:"id" gorm:"primaryKey"`
	UserID           uint      `json:"user_id"`
	Title            string    `json:"title"`
	Description      string    `json:"description"`
	SystemPrompt     string    `json:"system_prompt"`
	Icon             string    `json:"icon"`
	Color            string    `json:"color"`
	InitialMessage   string    `json:"initial_message"`
	InputPlaceholder string    `json:"input_placeholder"`
	SpeechLang       string    `json:"speech_lang"`
	SpeechRate       float64   `json:"speech_rate"`
	IsFavorite       bool      `json:"is_favorite"`
	CreatedAt        time.Time `json:"created_at"`
	UpdatedAt        time.Time `json:"updated_at"`
}

type CreateCustomTrainingRequest struct {
	Title            string  `json:"title" binding:"required"`
	Description      string  `json:"description"`
	SystemPrompt     string  `json:"system_prompt" binding:"required"`
	Icon             string  `json:"icon"`
	Color            string  `json:"color"`
	InitialMessage   string  `json:"initial_message"`
	InputPlaceholder string  `json:"input_placeholder"`
	SpeechLang       string  `json:"speech_lang"`
	SpeechRate       float64 `json:"speech_rate"`
}

type UpdateCustomTrainingRequest struct {
	Title            string  `json:"title"`
	Description      string  `json:"description"`
	SystemPrompt     string  `json:"system_prompt"`
	Icon             string  `json:"icon"`
	Color            string  `json:"color"`
	InitialMessage   string  `json:"initial_message"`
	InputPlaceholder string  `json:"input_placeholder"`
	SpeechLang       string  `json:"speech_lang"`
	SpeechRate       float64 `json:"speech_rate"`
}

File: frontend/src/service/api/custom-training.ts
import { request } from '../request';

export interface AIAgent {
  id: number;
  user_id: number;
  is_public: boolean;
  title: string;
  description: string;
  code: string;
  system_prompt: string;
  icon: string;
  color: string;
  initial_message: string;
  input_placeholder: string;
  speech_lang: string;
  speech_rate: number;
  is_favorite: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateAIAgentParams {
  title: string;
  description?: string;
  code?: string;
  system_prompt: string;
  icon?: string;
  color?: string;
  initial_message?: string;
  input_placeholder?: string;
  speech_lang?: string;
  speech_rate?: number;
}

export interface UpdateAIAgentParams {
  title?: string;
  description?: string;
  system_prompt?: string;
  icon?: string;
  color?: string;
  initial_message?: string;
  input_placeholder?: string;
  speech_lang?: string;
  speech_rate?: number;
}

export function fetchAIAgentList() {
  return request<AIAgent[]>({
    url: '/api/ai-agents',
    method: 'get'
  });
}

export function fetchAIAgentDetail(id: number) {
  return request<AIAgent>({
    url: `/api/ai-agents/${id}`,
    method: 'get'
  });
}

export function fetchCreateAIAgent(data: CreateAIAgentParams) {
  return request<AIAgent>({
    url: '/api/ai-agents',
    method: 'post',
    data
  });
}

export function fetchUpdateAIAgent(id: number, data: UpdateAIAgentParams) {
  return request({
    url: `/api/ai-agents/${id}`,
    method: 'put',
    data
  });
}

export function fetchDeleteAIAgent(id: number) {
  return request({
    url: `/api/ai-agents/${id}`,
    method: 'delete'
  });
}
